packages/server/src/api/controllers/query/index.ts

```diff
@@ -295,7 +297,7 @@ export async function preview(ctx: UserCtx) {
 }
 
 async function execute(
-  ctx: UserCtx<ExecuteQueryRequest, ExecuteQueryResponse | Row[]>,
+  ctx: UserCtx<ExecuteQueryRequest, ExecuteQueryResponse | any[]>,
```

Comment:
Can we execute this with something that is not a row?

Reply:
`Row[]` here was actually incorrect, as far as I can tell. Queries don't return `Row`s, they can return absolutely anything. It's raw SQL, we have little to no control over it. `Row` extends `Document`, and both types specify some optional fields that have types that may be wrong. It's entirely possible for a user to have a table with an `_id` column that's a date, for example.

Reply:
This is true, technically we could be a little more specific with a row response of `Record<string, any>` - it will always be an array of objects returned, even if the query returns say a string or number, we will wrap it in an object like `[{ value: 1 }]`.